components/places/src/import/common.rs

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::api::places_api::SyncConn;
+use crate::error::*;
+use rusqlite::named_params;
+use url::Url;
+
+pub mod sql_fns {
+    use crate::storage::URL_LENGTH_MAX;
+    use crate::types::Timestamp;
+    use rusqlite::{functions::Context, types::ValueRef, Result};
+    use url::Url;
+
+    #[inline(never)]
+    pub fn sanitize_timestamp(ctx: &Context<'_>) -> Result<Timestamp> {
+        let now = Timestamp::now();
+        Ok(if let Ok(ts) = ctx.get::<Timestamp>(0) {
+            if Timestamp::EARLIEST < ts && ts < now {
+                ts
+            } else {
+                now
+            }
+        } else {
+            now
+        })
+    }
+
+    #[inline(never)]
+    pub fn validate_url(ctx: &Context<'_>) -> Result<Option<String>> {
+        let val = ctx.get_raw(0);
+        let href = if let ValueRef::Text(s) = val {
+            s
+        } else {
+            return Ok(None);
+        };
+        if href.len() > URL_LENGTH_MAX {
+            return Ok(None);
+        }
+        if let Ok(url) = Url::parse(href) {
+            Ok(Some(url.into_string()))
+        } else {
+            Ok(None)
+        }
+    }
+
+    #[inline(never)]
+    pub fn is_valid_url(ctx: &Context<'_>) -> Result<Option<bool>> {
+        Ok(match ctx.get_raw(0) {
+            ValueRef::Text(s) if s.len() <= URL_LENGTH_MAX => Some(Url::parse(s).is_ok()),
+            // Should we do this?
+            // ValueRef::Null => None,
```

Comment:
Eh...I don't think so. It'd be consistent with how SQL functions handle `NULL`s, but I don't think we gain much from distinguishing between missing and invalid.